satellite/metainfo/metainfo.go

```diff
@@ -128,6 +129,18 @@ func (endpoint *Endpoint) SegmentInfoOld(ctx context.Context, req *pb.SegmentInf
 func (endpoint *Endpoint) CreateSegmentOld(ctx context.Context, req *pb.SegmentWriteRequestOld) (resp *pb.SegmentWriteResponseOld, err error) {
 	defer mon.Task()(&ctx)(&err)
 
+	if req.Expiration != nil {
+		exp, err := ptypes.Timestamp(req.Expiration)
+		if err != nil {
+			return nil, status.Errorf(codes.InvalidArgument, err.Error())
+		}
+
+		err = validateExpiration(ctx, exp)
+		if err != nil {
```

Comment:
`if expiration.Less(time.Now().Add(piece expiration thershold)) {`

Reply:
updated using exp.After(u Time) and cleaned up the code